qadence2_platforms/backends/_base_analog/functions.py

```diff
@@ -0,0 +1,212 @@
+
+        op: qutip.Qobj
+        arg: InputType
+
+        if expr.subspace:
+            native_ops: list[qutip.Qobj] = []
+            support_set: set = expr.subspace.subspace
+
+            for k in range(num_qubits):
+                if k in support_set:
+                    sub_num_qubits: int = cast(Support, expr.args[1]).max_index
+                    arg = cast(InputType, expr.args[0])
+                    op = cls._get_op(sub_num_qubits, arg)
+
+                else:
+                    op = cls.operators_mapping["I"]
+
+                native_ops.append(op)
+
+            return qutip.tensor(*native_ops)
+
+        arg = cast(InputType, expr.args[0])
+        op = cls._get_op(num_qubits, arg)
+        return qutip.tensor(*([op] * num_qubits))
+
+    @classmethod
+    def _arith_tensor_op(cls, num_qubits: int, expr: InputType) -> qutip.Qobj:
+        """
+        Use it for the arithmetic operations addition and multiplication that need to
+        have an extended Hilbert space compatible with `num_qubits`.
+
+        Args:
+            num_qubits (int): the number of qubits to create the qutip object to
+            expr (InputType): the input expression. Any qadence2-expressions expression
+                compatible object, with the same methods
+
+        Returns:
+            A QuTiP object with the Hilbert space compatible with `num_qubits`
+        """
+
+        subspace: set = cast(Support, expr.subspace).subspace
+        super_space: set = set(range(num_qubits))
+
+        if super_space.issuperset(subspace):
+            native_ops: list[qutip.Qobj] = []
+            arg_subspace: set = cast(Support, expr.args[1]).subspace
+
+            for k in range(num_qubits):
+                if k in arg_subspace:
+                    sub_num_qubits: int = cast(int, expr.args[1])
+                    arg: InputType = cast(InputType, expr.args[0])
+                    op: qutip.Qobj = cls._get_op(sub_num_qubits, arg)
+                    native_ops.append(op)
+                else:
+                    native_ops.append(cls.operators_mapping["I"])
+
+            return qutip.tensor(*native_ops)
+
+        raise ValueError(
+            f"subspace of the object ({subspace}) is bigger than the "
+            f"sequence space ({super_space})"
+        )
+
+    @classmethod
+    def _get_op(cls, num_qubits: int, op: InputType) -> qutip.Qobj | None:
+        """
+        Convert an expression into a native QuTiP object. A simple symbol,
+        a quantum operator, and an operation (addition, multiplication or
+        kron tensor) are valid objects.
+
+        Args:
+            num_qubits (int): the number of qubits to create the qutip object to
+            op (InputType): the input expression. Any qadence2-expressions expression
+                compatible object, with the same methods
+
+        Returns:
+            A QuTiP object with the Hilbert space compatible with `num_qubits`
+        """
+
+        if op.is_symbol is True:
```

Comment:
Would `op.is_symbol` suffice ?

Reply:
no. It's value matters, not if the object is empty or not